File: docs/guides/scala/src/main/scala/abTestImplementationChecklist.scala
import scala.concurrent.Future
import scala.concurrent.ExecutionContext.Implicits.global

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.search.SearchParamsObject

val getUserToken: String = {
  "" // Implement your logic here
}

def abTestImplementationChecklist(): Future[Unit] = {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  // Set the query and get the current user token
  val query = "User search query"
  val userToken = getUserToken

  // Set the searchParams
  var searchParams = SearchParamsObject(
    query = Some(query)
  )

  // Is the user token anonymous?
  if (userToken == null || userToken.isEmpty || userToken == "YOUR_ANONYMOUS_USER_TOKEN") {
    // Disable A/B testing for this request
    searchParams = searchParams.copy(enableABTest = Some(false))
  } else {
    // Set the user token to the current user token
    searchParams = searchParams.copy(userToken = Some(userToken))
  }

  // Perform the searchSingleIndex
  client
    .searchSingleIndex(
      indexName = "<YOUR_INDEX_NAME>",
      searchParams = Some(searchParams)
    )
    .map { result =>
      // SearchSingleIndex results
      println(result)
    }
    .recover { case err: Exception =>
      // SearchSingleIndex errors
      println(s"An error occurred: ${err.getMessage}")
    }
}

File: docs/guides/javascript/src/abTestImplementationChecklist.ts
import { algoliasearch } from 'algoliasearch';

import type { SearchParams } from 'algoliasearch';

const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

// Implement your logic here
const getUserToken = () => {
  return '';
};

// Set the searchParams and get the current user token
const searchParams: SearchParams = {
  query: 'User search query',
  enableABTest: true,
};
const userToken = getUserToken();

// Is the user token anonymous?
if (userToken === null || userToken === undefined || userToken === 'YOUR_ANONYMOUS_USER_TOKEN') {
  // Disable A/B testing for this request
  searchParams.enableABTest = false;
} else {
  // Set the user token to the current user token
  searchParams.userToken = userToken;
}

try {
  // Perform the searchSingleIndex
  const result = await client.searchSingleIndex({ indexName: 'indexName', searchParams: searchParams });
  // SearchSingleIndex results
  console.log(result);
} catch (err) {
  // SearchSingleIndex errors
  console.error(err);
}
